File: java/src/main/BC1533.java
import java.util.Scanner;

public class BC1533 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (sc.hasNextInt()) {
            int N = sc.nextInt();
            if (N == 0) {
                break;
            }

            int[] suspects = new int[N];
            for (int i = 0; i < N; i++) {
                suspects[i] = sc.nextInt();
            }

            int mostSuspect = Integer.MIN_VALUE;
            int secondMostSuspect = Integer.MIN_VALUE;

            for (int i = 0; i < N; i++) {
                if (suspects[i] > mostSuspect) {
                    secondMostSuspect = mostSuspect;
                    mostSuspect = suspects[i];
                } else if (suspects[i] > secondMostSuspect && suspects[i] < mostSuspect) {
                    secondMostSuspect = suspects[i];
                }
            }

            int iSecondMostSuspect = -1;
            for (int i = 0; i < N; i++) {
                if (suspects[i] == secondMostSuspect) {
                    iSecondMostSuspect = i + 1;
                    break;
                }
            }
            System.out.println(iSecondMostSuspect);
        }
        sc.close();
    }
}

File: c/bc1533.c
#include <stdio.h>
#include <limits.h>

int main() {
    int N;

    while (scanf("%d", &N) == 1 && N != 0) {
        int suspects[N];
        for (int i = 0; i < N; i++)
            scanf("%d", &suspects[i]);

        int most_suspect = INT_MIN;
        int second_most_suspect = most_suspect;
        int i_second_most_suspect = -1;

        for (int i = 0; i < N; i++) {
            if (suspects[i] > most_suspect) {
                second_most_suspect = most_suspect; // O antigo most_suspect se torna o second most_suspect
                most_suspect = suspects[i];  // O valor atual se torna o novo most_suspect
            } else if (suspects[i] > second_most_suspect && suspects[i] < most_suspect) {
                // Se o suspeito atual é mais suspeito que o second_most_suspect, mas menor que o most_suspect
                second_most_suspect = suspects[i];
            }
        }

        for (int i = 0; i < N; i++) {
            if (suspects[i] == second_most_suspect) {
                i_second_most_suspect = i+1;
                break;
            }
        }
        printf("%d\n", i_second_most_suspect);
    }

    return 0;
}